frontend/src/components/LocaleSwitcher.tsx

```diff
@@ -0,0 +1,45 @@
+import type { FallbackLng } from "i18next";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "src/components/ui/select";
+import { toast } from "src/components/ui/use-toast";
+import i18n, { supportedLocales } from "src/i18n/i18nConfig";
+
+export default function LocaleSwitcher() {
+  const { t } = useTranslation("components", {
+    keyPrefix: "locale_switcher",
+  });
+  const fallbackLng = i18n.options.fallbackLng?.[0 as keyof FallbackLng];
+  const [dropdownLang, setDropdownLang] = useState(
+    i18n.language || fallbackLng
+  );
+
+  const onLanguageChange = async (newLanguage: string) => {
+    if (dropdownLang !== newLanguage) {
+      setDropdownLang(newLanguage);
+      i18n.changeLanguage(newLanguage);
+      toast({ title: t("success") });
+    }
+  };
+
+  return (
+    <Select value={dropdownLang} onValueChange={onLanguageChange}>
+      <SelectTrigger className="w-[150px]">
+        <SelectValue placeholder="Language" />
```

Comment:
aNot sure if we need a translation for Language too, if yes where should it go? In components or common?

Reply:
I put it in the component for now